src/utils/references.ts

```diff
@@ -44,3 +45,47 @@ export function stringifyReferences (paths: string[], cwd: string): string {
 export function toReference (path: string, cwd: string): string {
   return `/// <reference path="${normalizeSlashes(isAbsolute(path) ? relative(cwd, path) : normalize(path))}" />`
 }
+
+export enum DependencyTarget {
+  Main,
+  Browser
+}
+
+export enum DependencyType {
+  Ambient,
+  External
+}
+
+export interface DependencyInfo {
+  target: DependencyTarget
+  type: DependencyType
+  name: string
+}
+
+export function parseReferencePath(path: string): DependencyInfo {
+  const referencePathRegexPattern =
```

Comment:
> is it possible for the name to contain path.sep?

Yes, it can. Check out the function that creates the paths in `path.ts`, it just joins so anything with a slash (or backslash) become segments of the path.

> is it safe to assume that it will always be index.d.ts?

Safe to assume, yes. See the same function I mentioned above. It's generating the paths where stuff is placed. I'd just make the reverse of that functions logic. https://github.com/typings/core/blob/master/src/utils/path.ts#L166